### PR TITLE
Template

### PR DESCRIPTION
**Pull request by pesser**
- provide unified starting templates for tensorflow and pytorch
- introduces LambdaCheckpointHook for easy and generic saving and restoring
- better imports: `from edflow import PyHookedIterator, ...`, see `edflow/__init__.py`
- update `README.md`: not sexy yet, but more helpful with explicit getting started guide
- extend evaluation pipeline to look only into specified keypath
- add TemplateEvalHook to disable writing output if eval_op returns nothing, i.e. use of eval pipeline is optional in TemplateIterator
- introduces LoggingHook: A minimal logger which only understands `scalars` and `images` but can be easily extended by adding custom `handlers`
- introduces TemplateIterator which provides reasonable defaults for both training and evaluation.

### CODE REVIEW

**Comment by pesser** [comment]
the `test_tf_nn.py` test fails. it fails due to `import scikit-image`. replacing the test with this single line fails as well. The error is `dlopen: Cannot load any more object with static TLS`.
Hard to figure out what's going on https://github.com/scikit-learn/scikit-learn/issues/14485
Removing the `import` statements from `edflow/__init__.py` fixes it, but I really want to simplify the import statements. Also, the test fails on travis ci but not on our machines.

**Comment by pesser** [comment]
running `python -c "import skimage"` before running the tests fixes the problem. Digging deeper, the problem seems to be `.libs/libgomp-3300acd3.so.1.0.0` shipped with `scikit-image`. It uses static TLS, which it probably should not do https://bugzilla.redhat.com/show_bug.cgi?id=1124987

File: edflow/eval/pipeline.py [hunk 1]
```diff
@@ -388,7 +424,7 @@ def _delget(d, k):
 
 def save_example(savepath, datum):
     """Manages the writing process of a single datum: (1) Determine type,
-    (2) Choos saver, (3) save.
+    (2) Choose saver, (3) save.
 
     Args:
```
Comment (ArWeHei) [inline edflow/eval/pipeline.py]: wrong docstring format :(

File: edflow/hooks/logging_hooks/minimal_logging_hook.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+from edflow.iterators.batches import plot_batch
+import os
+
+
+class LoggingHook(Hook):
+    """Minimal implementation of a logging hook. Can be easily extended by
+    adding handlers."""
+
+    def __init__(self, paths, interval, root_path):
+        """Args:
```
Comment (ArWeHei) [inline edflow/hooks/logging_hooks/minimal_logging_hook.py]: Wrong docstring format again

**Review by ArWeHei — CHANGES_REQUESTED** [review]
I believe that the test introduced in #131 should break when this PR is applied. Still, we should start keeping the docstrings consistent.

**Review by ArWeHei — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**